asreviewcontrib/insights/utils.py

```diff
@@ -1,8 +1,33 @@
 import numpy as np
+from asreview import open_state
+from asreview.state import SQLiteState
 
 
 def get_labels(state_obj, priors=False):
+    """Get the labels from state file(s).
 
+    Parameters
+    ----------
+    state_obj : Union[SQLiteState, list[SQLiteState]]
+        Single state object, or list of multiple state objects.
+    priors : bool, optional
+        Include the prior labels, by default False
+
+    Returns
+    -------
+    list or list of lists
+        List of labels if state_obj is a single state. List of lists of labels
+        if state_obj is a list of states.
+    """
+    print(type(state_obj))
+    if isinstance(state_obj, SQLiteState):
+        return _get_labels(state_obj, priors)
+    else:
+        return [_get_labels(single_state, priors) for single_state in state_obj]
+
```

Comment:
I prefer recursive functions 

```suggestion
    if isinstance(state_obj, SQLiteState):
        n_records = state_obj.n_records
    
        # get the labels
        labels = state_obj.get_labels(priors=priors).to_list()
        # if less labels than records, check if all labels available
        if len(labels) < n_records:
            labels = labels + np.zeros(n_records - len(labels)).tolist()
    
        return labels
    else:
        return [get_labels(single_state, priors) for single_state in state_obj]

```

Something like this